src/common/users.ts

```diff
@@ -339,3 +354,58 @@ export const getUserReadingRank = async (
     tags,
   };
 };
+
+export enum Day {
+  Sunday,
+  Monday,
+  Tuesday,
+  Wednesday,
+  Thursday,
+  Friday,
+  Saturday,
+}
+
+export const Weekends = [Day.Sunday, Day.Saturday];
+export const FREEZE_DAYS_IN_A_WEEK = Weekends.length;
+export const MISSED_LIMIT = 1;
+
+export const clearUserStreak = async (
+  con: DataSource | EntityManager,
+  userId,
+): Promise<boolean> => {
+  const result = await con
+    .getRepository(UserStreak)
+    .update({ userId }, { currentStreak: 0, updatedAt: new Date() });
+
+  return result.affected > 0;
+};
+
+export const checkAndClearUserStreak = async (
+  ctx: Context,
+  info: GraphQLResolveInfo,
+  streak: GQLUserStreakTz,
+): Promise<boolean> => {
+  const { lastViewAtTz: lastViewAt, timezone: timeZone } = streak;
+
+  if (!lastViewAt) {
+    return false;
+  }
+
+  const now = new Date();
+  const timeZonedToday = now.toLocaleDateString('en', { timeZone });
+  const today = new Date(timeZonedToday);
```

Comment:
Potential improvement - move this to a utility function.

Reply:
Yeah was thinking of that too. I'll move it today.